OOPHomework14January/src/Store.java

```diff
@@ -0,0 +1,292 @@
+        } else {
+            System.out.println("The product " + productName +
+                    " was not added to the Store!!!");
+            System.out.println("The id given to " + productName +
+                    " is already given to another product.");
+        }
+    }
+
+
+    /**
+     * method to sell a product
+     *
+     * @param id       - product's id that will be sold
+     * @param quantity - the quantity we want to sell
+     */
+    public boolean sellAnimalProduct(int id, int quantity) {
+        boolean found = false;
+        for (AnimalProduct animalProduct : arrayAnimalProduct) {
+            // if in the array of animal products exists a product with this id, and we have the appropriate quantity
+            if ((animalProduct.getId() == id) && (animalProduct.getStock() >= quantity)) {
+                // the product will be added in an array of orders
+                arrayOrders.add(new Order(localDate(), id, quantity));
+                // set the new stock of the product after selling
+                animalProduct.setStock(animalProduct.getStock() - quantity);
+                found = true;
+            } else if ((animalProduct.getId() == id) && (animalProduct.getStock() < quantity)) {
+                // if the product exists, but there is no stock
+                System.out.println("Insufficient stock for the product!!!");
+                found = true;
+            }
+        }
+        return found;
+    }
+
+    /**
+     * method to sell a vegetal product
+     *
+     * @param id       - product's id that will be sold
+     * @param quantity - the quantity we want to sell
+     */
+    public  boolean sellVegetalProduct(int id, int quantity) {
+        boolean found = false;
+        for (VegetalProduct vegetalProduct : arrayVegetalProduct) {
+            // if in the array of animal products exists a product with this id, and we have the appropriate quantity
+            if ((vegetalProduct.getId() == id) && (vegetalProduct.getStock() >= quantity)) {
+                // the product will be added in an array of orders
+                arrayOrders.add(new Order(localDate(), id, quantity));
+                // set the new stock of the product after selling
+                vegetalProduct.setStock(vegetalProduct.getStock() - quantity);
+                found = true;
+            } else if ((vegetalProduct.getId() == id) && (vegetalProduct.getStock() < quantity)) {
+                // if the product exists, but there is no stock
+                System.out.println("Insufficient stock for the product!!!");
+                found = true;
+            }
+        }
+        return found;
+    }
+
+    /**
+     * method to avoid registering a duplicate id
+     *
+     * @param id - the id we want to check
+     * @return true - if the id is already given to another product; and false - otherwise
+     */
+    private boolean duplicateAnimalID(int id) {
```

Comment:
again, duplicateId that checks the List<Product> 
no need to know here if you're dealing with animal or vegetable